matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt

```diff
@@ -105,16 +105,9 @@ internal class TestMatrix constructor(context: Context, matrixConfiguration: Mat
             }
         }
 
-        fun getInstance(context: Context): TestMatrix {
-            if (isInit.compareAndSet(false, true)) {
-                val appContext = context.applicationContext
-                if (appContext is MatrixConfiguration.Provider) {
-                    val matrixConfiguration = (appContext as MatrixConfiguration.Provider).providesMatrixConfiguration()
-                    instance = TestMatrix(appContext, matrixConfiguration)
-                } else {
-                    throw IllegalStateException("Matrix is not initialized properly." +
-                            " You should call Matrix.initialize or let your application implements MatrixConfiguration.Provider.")
-                }
+        fun getInstance(): TestMatrix {
```

Comment:
might be worth deleting these in favour of syncing with the production Matrix creation and only providing a `createInstance`, that's my bad for not updating this area as well https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android%2Fsrc%2Fmain%2Fjava%2Forg%2Fmatrix%2Fandroid%2Fsdk%2Fapi%2FMatrix.kt#L106

Reply:
I had a look at this; the problem I ran into was that we can only initialize it once; but the CommonTestHelper is created multiple times, so creates multiple TestMatrix  instances - so we'd need somewhere to store a static instance of it across all tests, so keeping it in TestMatrix seemed reasonable - in that sense the TestMatrix class is the thing that manages having a single instance of itself.

(I might be missing tricks though)

Reply:
I haven't dived too deep into the integration tests and this probably deserves an entirely separate discussion/issue :sweat_smile:  but I would have expected our test instance to not use a singleton state

Reply:
So the error came from creating an additional `androidx.work.WorkManager` on each initialization of TestMatrix. So I'm not sure quite whether we want to share one of those or if we want to tear it down on test finishing or ... 

TL;DR: this could be fixed but I don't want to dive into it to remove the errors as warnings flag.